.github/workflows/publish-to-npm.yaml

```diff
@@ -0,0 +1,23 @@
+# Publish package to npm whenever a release is pushed.
+# See https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+name: Publish to npm
+on:
+  workflow_call:
+    inputs:
+      secrets:
+        NPM_TOKEN:
+          description: 'A token to enable this workflow to publish to npm.'
+          required: true
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '20.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm ci
```

Comment:
In general — I’ve updated things to use [clean install](https://docs.npmjs.com/cli/v9/commands/npm-ci) so that we install from `package-lock.json` for consistency.